keras/backend/theano_backend.py

```diff
@@ -589,31 +589,19 @@ def any(x, axis=None, keepdims=False):
     """Bitwise reduction (logical OR).
     """
     y = T.any(x, axis=axis, keepdims=keepdims)
-    if hasattr(x, '_keras_shape'):
-        if axis is None:
-            y._keras_shape = (1,) * len(x._keras_shape) if keepdims else (1,)
-        else:
-            if isinstance(axis, int):
-                axis_list = [axis]
-            else:
-                axis_list = list(set(int(a) for a in axis))
-            keras_shape_list = list(x._keras_shape)
-            if keepdims:
-                for a in axis_list:
-                    keras_shape_list[a] = 1
-            else:
-                for a in axis_list[::-1]:
-                    keras_shape_list.pop(a)
-                if not keras_shape_list:
-                    keras_shape_list = (1,)
-            y._keras_shape = tuple(keras_shape_list)
+    y = _set_keras_shape_any_all(x, y, axis, keepdims)
     return y
 
 
 def all(x, axis=None, keepdims=False):
     """Bitwise reduction (logical AND).
     """
     y = T.all(x, axis=axis, keepdims=keepdims)
+    y = _set_keras_shape_any_all(x, y, axis, keepdims)
+    return y
+
+
+def _set_keras_shape_any_all(x, y, axis, keepdims):
```

Comment:
`_set_keras_shape_for_reduction` would arguably be a better name. This can be reused for any reduction op (sum, etc).

Reply:
Naming things is hard. Let me change that.